apps/challenges/views.py

```diff
@@ -2740,3 +2742,36 @@ def manage_worker(request, challenge_pk, action):
             response_data = {"action": "Failure", "error": message}
 
     return Response(response_data, status=status.HTTP_200_OK)
+
+
+@api_view(["GET"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_presigned_url_for_annotations(
+    request, challenge_pk, challenge_phase_pk
+):
+    if not is_user_a_host_of_challenge(request.user, challenge_pk):
+        response_data = {
+            "error": "Sorry, you are not authorized for uploading an annotation file."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    challenge_phase = get_challenge_phase_model(challenge_phase_pk)
+
+    file_ext = os.path.splitext(request.data["file_name"])[-1]
+    file_name = "test_annotations/presigned_url_files/challenge_phase_{}/{}{}".format(
```

Comment:
We're just hardcoding the logical parent folders of the file. The file name itself is random. This is consistent with the way the ChallengePhase model stores the files. The only additional thing I've specified is a `presigned_url_files` parent folder.
What do you think?

Reply:
I still didn't get this or maybe I missing something here -- When you will call the save method would that automatically put the file in the respective folder? Why do we need a hardcoded name?

Reply:
It'll automatically put the dummy file in the default folder like so: 
`test_annotation = models.FileField(upload_to=RandomFileName("test_annotations"), default=False)`.

What's being done here is the following:
1. Dummy file is being uploaded automatically like usual (back when they create the challenge.)
2. Now, when uploading through CLI: This API will generate the presigned url to upload the new annotation file to a different `presigned_url` subfolder within the `test_annotation` folder. (I guess I can remove the` challenge_phase_{}`) folder to have just one additional new folder.
3. After this, the file_name for the current dummy file should be changed so that when the worker fetches the file's url, it can point to the new annotation file. (And this is what works to change the name of a file in Django: `challenge_phase.test_annotation = file_name`)

Are you saying upload the actual annotation file using the Presigned to the same location as the dummy file with the same `uuid.uuid4()` name?  (That way there's no need to change the file name in the ChallengePhase object since the new annotation file will be superimposed on the previous dummy file.)